Sources/UI/Chat/Chat View Controller/ChatViewController+Composer.swift

```diff
@@ -466,34 +466,33 @@ extension ChatViewController {
             do {
                 let pickedImage = try result.get()
                 
-                guard let fileURL = pickedImage.fileURL else {
-                    ClientLogger.log("📁", "File URL cannot be determined for file named: \(pickedImage.fileName)")
+                guard let presenter = self.presenter, let image = pickedImage.image else {
                     return
                 }
                 
-                let fileSizeResource = try fileURL.resourceValues(forKeys: [.fileSizeKey])
+                let extraData = presenter.imageAttachmentExtraDataCallback?(pickedImage.fileURL,
+                                                                            image,
+                                                                            pickedImage.isVideo,
+                                                                            presenter.channel)
                 
-                if let fileSize = fileSizeResource.fileSize,
-                   fileSize >= 20 * 1_048_576 { // 20 MB Upload limit
+                let uploaderItem = try UploadingItem(channel: presenter.channel, pickedImage: pickedImage, extraData: extraData)
+                
+                guard let data = uploaderItem.data, data.count < 20 * 1024 * 1024 else {
```

Comment:
I thought you'll update a file size inside `UploadingItem`. So we won't need to read full file data (for URL case) to get the size of it.

Reply:
`UploadingItem` will parse the `data` for uploading anyway, we can use `data.count` to get the size easily.